stetho/src/main/java/com/facebook/stetho/inspector/elements/android/ActivityTracker.java

```diff
@@ -0,0 +1,209 @@
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.stetho.inspector.elements.android;
+
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.app.Application;
+import android.database.Observable;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.Looper;
+import com.facebook.stetho.common.Util;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.GuardedBy;
+import javax.annotation.concurrent.NotThreadSafe;
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Tracks which {@link Activity} instances have been created and not yet destroyed in creation
+ * order for use by Stetho features.  Note that automatic tracking is not available for
+ * all versions of Android but it is possible to manually track activities using the {@link #add}
+ * and {@link #remove} methods exposed below.  Be aware that this is an easy opportunity to
+ * cause serious memory leaks in your application however.  Use with caution.
+ * <p/>
+ * Most callers can and should ignore this class, though it is necessary if you are implementing
+ * Activity tracking pre-ICS.
+ */
+@NotThreadSafe
+public class ActivityTracker {
+  private static final ActivityTracker sInstance = new ActivityTracker();
+
+  @GuardedBy("Looper.getMainLooper()")
+  private final ArrayList<Activity> mActivities = new ArrayList<>();
+  private final List<Activity> mActivitiesUnmodifiable = Collections.unmodifiableList(mActivities);
+
+  @GuardedBy("ActivityTrackerObservable.mObservers")
+  private final ActivityTrackerObservable mObservable = new ActivityTrackerObservable();
+
+  @Nullable
+  private AutomaticTracker mAutomaticTracker;
+
+  public static ActivityTracker get() {
+    return sInstance;
+  }
+
+  public void registerListener(Listener listener) {
+    mObservable.registerObserver(listener);
+  }
+
+  public void unregisterListener(Listener listener) {
+    mObservable.unregisterObserver(listener);
+  }
+
+  /**
+   * Start automatic tracking if we are running on ICS+.
+   *
+   * @return Automatic tracking has been started.  No need to manually invoke {@link #add} or
+   *     {@link #remove} methods.
+   */
+  public boolean beginTrackingIfPossible(Application application) {
+    if (mAutomaticTracker == null) {
+      AutomaticTracker automaticTracker =
+          AutomaticTracker.newInstance(application, this /* tracker */);
+      if (automaticTracker != null) {
+        automaticTracker.register();
+        mAutomaticTracker = automaticTracker;
+        return true;
+      }
+    }
+    return false;
+  }
+
+  public boolean endTracking() {
```

Comment:
I thought it might be convenient to know if tracking was enabled when you called end.  It really doesn't matter I suppose, this method is never even called :)